File: Native/Android/LunarConsole/lunarConsole/src/androidTest/java/spacemadness/com/lunarconsole/concurrent/ImmediateDispatchQueue.java
package spacemadness.com.lunarconsole.concurrent;

public class ImmediateDispatchQueue extends DispatchQueue {
    public ImmediateDispatchQueue() {
        super("immediate");
    }

    @Override
    protected void schedule(DispatchTask task, long delay) {
        task.execute();
    }

    @Override
    public void stop() {
    }

    @Override
    public boolean isCurrent() {
        return false;
    }
}

File: Native/Android/LunarConsole/lunarConsole/src/test/java/spacemadness/com/lunarconsole/concurrent/ImmediateDispatchQueue.kt
package spacemadness.com.lunarconsole.concurrent

class ImmediateDispatchQueue(
        private val dispatchImmediately: Boolean = false
) : DispatchQueue("Immediate Queue") {
    private val tasks = mutableListOf<DispatchTask>()

    override fun isCurrent() = true

    override fun stop() {
    }

    override fun schedule(task: DispatchTask, delay: Long) {
        if (dispatchImmediately) {
            task.execute()
        } else {
            tasks.add(task)
        }
    }

    fun dispatchTasks() {
        tasks.forEach { task -> task.run() }
        tasks.clear()
    }
}